### ISSUE TITLE
Cannot deserialize model: mismatched structure

### ISSUE THREAD

**Issue by jacob-jeppesen** [explosion/spaCy#8566]
## How to reproduce the behaviour

spaCy .from_bytes() seem to be broken when it comes to using transformers. It works when I use non-transformer models such as 'en_core_web_sm' but not for 'en_core_web_trf':

```python
#%% Setup
import spacy

trf = spacy.load('en_core_web_trf')
config = trf.config
trf_bytes = trf.to_bytes()

#%% Setup nlp object
lang_cls = spacy.util.get_lang_class(config['nlp']['lang'])
nlp = lang_cls.from_config(config)
nlp.from_bytes(trf_bytes)

#%% Test things
sample = 'On June 26th 1963 John F. Kennedy gave a speech in Berlin.'

for entity in nlp(sample).ents:
    print(entity.text, entity.label_)
```
Returns:
```
Traceback (most recent call last):
  File ".\saga_bytes1.py", line 11, in <module>
    nlp.from_bytes(trf_bytes)
  File "C:\ProgramData\Anaconda3\lib\site-packages\spacy\language.py", line 1945, in from_bytes
    util.from_bytes(bytes_data, deserializers, exclude)
  File "C:\ProgramData\Anaconda3\lib\site-packages\spacy\util.py", line 1127, in from_bytes
    return from_dict(srsly.msgpack_loads(bytes_data), setters, exclude)
  File "C:\ProgramData\Anaconda3\lib\site-packages\spacy\util.py", line 1149, in from_dict
    setter(msg[key])
  File "C:\ProgramData\Anaconda3\lib\site-packages\spacy\language.py", line 1942, in <lambda>
    deserializers[name] = lambda b, proc=proc: proc.from_bytes(
  File "spacy\pipeline\trainable_pipe.pyx", line 295, in spacy.pipeline.trainable_pipe.TrainablePipe.from_bytes
  File "C:\ProgramData\Anaconda3\lib\site-packages\spacy\util.py", line 1127, in from_bytes
    return from_dict(srsly.msgpack_loads(bytes_data), setters, exclude)
  File "C:\ProgramData\Anaconda3\lib\site-packages\spacy\util.py", line 1149, in from_dict
    setter(msg[key])
  File "spacy\pipeline\trainable_pipe.pyx", line 286, in spacy.pipeline.trainable_pipe.TrainablePipe.from_bytes.load_model
  File "C:\ProgramData\Anaconda3\lib\site-packages\thinc\model.py", line 529, in from_bytes
    return self.from_dict(msg)
  File "C:\ProgramData\Anaconda3\lib\site-packages\thinc\model.py", line 546, in from_dict
    raise ValueError("Cannot deserialize model: mismatched structure")
ValueError: Cannot deserialize model: mismatched structure
```
While 'en_core_web_sm' returns:
```
June 26th DATE
1963 DATE
John F. Kennedy PERSON
Berlin GPE
```

## Your Environment

* Operating System: Microsoft Windows Server 2019 Datacenter
* Python Version Used: 3.8.5 64-bit ('base': conda)
* spaCy Version Used: 3.0.6 / spacy-transformers 1.0.3 / en-core-web-trf 3.0.0 / thinc 8.0.6
* Environment Information:

**Comment by svlandeg** [comment]
Thanks for the report and the reproducible snippet, that's always highly appreciated :-)

Ugh, this does look like a bug. I think I know where the problem is, I've been looking into something similar recently. I'll need to dig into this a bit.

**Comment by svlandeg** [comment]
Confirmed that this is the same bug as what I've described [here](https://github.com/explosion/spaCy/issues/8319#issuecomment-865087843).

Your code snippet
```
...
nlp = lang_cls.from_config(config)
nlp.from_bytes(trf_bytes)
...
```
will work when ensuring that the transformer is initialized BEFORE calling `from_bytes`. That's not the API you'd expect so we still need to fix this. I'll look into it further. 

But for now, if it helps you to continue working on this, you can do something like this:
```
nlp = lang_cls.from_config(config)
get_examples = lambda: [Example.from_dict(Doc(nlp.vocab, words=["x", "y", "z"]), {})]
nlp.get_pipe("transformer").initialize(get_examples, nlp=nlp)
nlp.from_bytes(trf_bytes)
```
To initialize a specific component, we require a function that creates `Example` objects. This code just creates a dummy example, similar as to how `nlp.initialize` would do it if it doesn't receive a `get_examples` argument.

With this addition, I get:
```
June 26th 1963 DATE
John F. Kennedy PERSON
Berlin GPE
```

**Comment by jacob-jeppesen** [comment]
It is a bit of a strange workaround but it did the trick!

Thanks for the help and quick response time 👍

**Comment by svlandeg** [comment]
We're going to get this properly fixed from `spacy-transformers` 1.1 onwards: https://github.com/explosion/spacy-transformers/pull/285

In case you want to try it out, the current bugfix is on the `develop` branch.

Thanks again for your report!
